### PR TITLE
add SVTR LCNet PP-OCRv3 CH model

### PR DESCRIPTION
**Pull request by tonytonglt**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: mindocr/models/heads/rec_sar_head.py [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+        '''
+        if img_metas is not None:
+            assert ops.shape(img_metas[0])[0] == ops.shape(feat)[0]
+
+        valid_width_masks = None
+        if img_metas is not None and self.mask:
+            valid_width_masks = img_metas[-1]
+
+        lab_embedding = self.embedding(label)
+        # bsz * seq_len * emb_dim
```
Comment (panshaowu) [inline mindocr/models/heads/rec_sar_head.py:283]: 无需使用的代码直接删除，不应用注释的方式保留。下同

File: tools/infer/text/postprocess.py [hunk 2]
```diff
@@ -49,12 +49,19 @@ def __init__(self, task="det", algo="DB", **kwargs):
             # TODO: update character_dict_path and use_space_char after CRNN trained using en_dict.txt released
             if algo.startswith("CRNN") or algo.startswith("SVTR"):
                 # TODO: allow users to input char dict path
-                dict_path = "mindocr/utils/dict/ch_dict.txt" if algo == "CRNN_CH" else None
-                postproc_cfg = dict(
-                    name="RecCTCLabelDecode",
-                    character_dict_path=dict_path,
-                    use_space_char=False,
-                )
+                dict_path = "mindocr/utils/dict/ch_dict.txt" if algo == "CRNN_CH" or algo == "SVTR_PPOCRv3_CH" else None
```
Comment (panshaowu) [inline tools/infer/text/postprocess.py:52]: if algo in ["CRNN_CH", "SVTR_PPOCRv3_CH"]
方便后续拓展

File: configs/rec/svtr/README_CN_PP-OCRv3.md [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+ - SVTR_LCNet：轻量级文本识别网络；
+ - GTC：Attention指导CTC训练策略；
+ - TextConAug：挖掘文字上下文信息的数据增广策略；
+ - TextRotNet：自监督的预训练模型；
+ - UDML：联合互学习策略；
+ - UIM：无标注数据挖掘方案。
+
+## 2. 权重转换
+
+如您已经有采用PaddleOCR训练好的PaddlePaddle模型，想在MindOCR下直接进行推理或进行微调续训，您可以对训练好的模型转换为MindSpore格式的ckpt文件。
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply configs/rec/svtr/README_CN_PP-OCRv3.md]: 已修改，同步修改DBNet README

File: mindocr/data/transforms/svtr_transform.py [hunk 4]
```diff
@@ -409,3 +413,694 @@ def __call__(self, data):
...
+        else:
+            self.gtc_encode = eval(gtc_encode)(max_text_length, character_dict_path, use_space_char, **kwargs)
+
+    def __call__(self, data):
+        data_ctc = copy.deepcopy(data)
+        data_gtc = copy.deepcopy(data)
+        data_out = dict()
+        data_out["img_path"] = data.get("img_path", None)
+        data_out["image"] = data["image"]
+        ctc = self.ctc_encode.__call__(data_ctc)
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply mindocr/data/transforms/svtr_transform.py:560]: 此处由于会预生成预处理transform_pipeline的实例列表，后续对每个batch使用`transform(data)`进行分步预处理，因此需要写为__call__调用形式，参考代码https://github.com/mindspore-lab/mindocr/blob/main/mindocr/data/transforms/transforms_factory.py#L68

File: mindocr/data/transforms/svtr_transform.py [hunk 5]
```diff
@@ -409,3 +413,694 @@ def __call__(self, data):
...
+    # generate valid_width_mask
+    width_downsampled = int(image_shape[-1] * width_downsample_ratio)
+    valid_width_mask = np.full([1, width_downsampled], 1).astype("int32")
+    valid_width = min(width_downsampled, int(width_downsampled * valid_ratio))
+    valid_width_mask[:, valid_width:] = 0
+
+    return padding_im, valid_ratio, valid_width_mask
+
+
+def resize_norm_img(
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply mindocr/data/transforms/svtr_transform.py:1031]: 此处由于需要将参考代码中valid_ratio转换为valid_width_mask以规避动态长度tensor切片引入的动态shape，有一定特殊性

File: mindocr/models/backbones/rec_svtr_enhance.py [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+        self.act = act()
+
+    def construct(self, inputs):
+        out = self.conv(inputs)
+        out = self.norm(out)
+        out = self.act(out)
+        return out
+
+
+class SEModule(nn.Cell):
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply mindocr/models/backbones/rec_svtr_enhance.py]: 已修改，复用utils下SEModule

File: mindocr/models/heads/rec_multi_head.py [hunk 7]
```diff
@@ -0,0 +1,455 @@
...
+        super().__init__()
+        self.result = None
+        self.sigmoid = nn.Sigmoid()
+
+    def construct(self, x):
+        result = x * self.sigmoid(x)
+        return result
+
+
+class ConvBNLayer(nn.Cell):
```
[the remark this answers is not in the document]
Comment (tonytonglt) [reply mindocr/models/heads/rec_multi_head.py:24]: 此处由于需要判断是否使用自定义的Swith激活函数，有一定特殊性；另外通常从mindcv引用模型作为backbone使用，较少引入单一功能模块

File: mindocr/models/heads/rec_sar_head.py (hunk 1 above)
Comment (tonytonglt) [reply mindocr/models/heads/rec_sar_head.py]: 此处主要为tensor shape的注释，便于代码阅读和调试，已修改为代码后注释形式

File: tools/infer/text/postprocess.py (hunk 2 above)
Comment (tonytonglt) [reply tools/infer/text/postprocess.py]: 已修改

**Review by panshaowu — APPROVED** [review]

**Review by VictorHe-1 — APPROVED** [review]

**Pull request merged**